src/Symfony/Component/Lock/Lock.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock;
+
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Store\BlockingStoreInterface;
+use Symfony\Component\Lock\Store\ExpirableStoreInterface;
+
+/**
+ * Lock is the default implementation of the LockInterface.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class Lock implements LockInterface
+{
+    use LoggerAwareTrait;
+
+    private $store;
+    private $key;
+
+    public function __construct(Key $key, StoreInterface $store)
+    {
+        $this->store = $store;
+        $this->key = $key;
+
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function acquire($blocking = false, $ttl = 300.0)
```

Comment:
Finally move the parameter in the constructor (as suggested) and remove it from the refresh method too (because this method serves the same purpose).